src/ripper/ripcddialog.cpp

```diff
@@ -0,0 +1,299 @@
+#include <QMessageBox>
+#include <QSettings>
+
+#include "config.h"
+#include "core/logging.h"
+#include "core/tagreaderclient.h"
+#include "core/utilities.h"
+#include "ripper/ripper.h"
+#include "ui_ripcddialog.h"
+#include "transcoder/transcoder.h"
+#include "transcoder/transcoderoptionsdialog.h"
+#include "ui/iconloader.h"
+
+namespace {
+bool ComparePresetsByName(const TranscoderPreset& left,
+                          const TranscoderPreset& right) {
+  return left.name_ < right.name_;
+}
+
+const int kCheckboxColumn = 0;
+const int kTrackNumberColumn = 1;
+const int kTrackTitleColumn = 2;
+}
+
+const char* RipCDDialog::kSettingsGroup = "Transcoder";
+const int RipCDDialog::kMaxDestinationItems = 10;
+
+RipCDDialog::RipCDDialog(QWidget* parent)
+    : QDialog(parent),
+      ui_(new Ui_RipCDDialog),
+      ripper_(new Ripper(this)),
+      working_(false) {
+  // Init
+  ui_->setupUi(this);
+
+  // Set column widths in the QTableWidget.
+  ui_->tableWidget->horizontalHeader()->setResizeMode(
+      kCheckboxColumn, QHeaderView::ResizeToContents);
+  ui_->tableWidget->horizontalHeader()->setResizeMode(
+      kTrackNumberColumn, QHeaderView::ResizeToContents);
+  ui_->tableWidget->horizontalHeader()->setResizeMode(kTrackTitleColumn,
+                                                      QHeaderView::Stretch);
+
+  // Add a rip button
+  rip_button_ = ui_->button_box->addButton(tr("Start ripping"),
+                                           QDialogButtonBox::ActionRole);
+  cancel_button_ = ui_->button_box->button(QDialogButtonBox::Cancel);
+  close_button_ = ui_->button_box->button(QDialogButtonBox::Close);
+
+  // Hide elements
+  cancel_button_->hide();
+  ui_->progress_group->hide();
+
+  connect(ui_->select_all_button, SIGNAL(clicked()), SLOT(SelectAll()));
+  connect(ui_->select_none_button, SIGNAL(clicked()), SLOT(SelectNone()));
+  connect(ui_->invert_selection_button, SIGNAL(clicked()),
+          SLOT(InvertSelection()));
+  connect(rip_button_, SIGNAL(clicked()), SLOT(ClickedRipButton()));
+  connect(cancel_button_, SIGNAL(clicked()), ripper_, SLOT(Cancel()));
+  connect(close_button_, SIGNAL(clicked()), SLOT(hide()));
+
+  connect(ui_->options, SIGNAL(clicked()), SLOT(Options()));
+  connect(ui_->select, SIGNAL(clicked()), SLOT(AddDestination()));
+
+  connect(ripper_, SIGNAL(Finished()), SLOT(Finished()));
+  connect(ripper_, SIGNAL(Cancelled()), SLOT(Cancelled()));
+  connect(ripper_, SIGNAL(ProgressInterval(int, int)),
+          SLOT(SetupProgressBarLimits(int, int)));
+  connect(ripper_, SIGNAL(Progress(int)), SLOT(UpdateProgressBar(int)));
+
+  setWindowTitle(tr("Rip CD"));
+  AddDestinationDirectory(QDir::homePath());
+
+  // Get presets
+  QList<TranscoderPreset> presets = Transcoder::GetAllPresets();
+  qSort(presets.begin(), presets.end(), ComparePresetsByName);
+  for (const TranscoderPreset& preset : presets) {
+    ui_->format->addItem(
+        QString("%1 (.%2)").arg(preset.name_, preset.extension_),
```

Comment:
`.arg(a).arg(b)` is preferred as it avoids future bugs when formatting numbers due to the multiple overloads of `arg()`